src/main/kotlin/io/sip3/salto/ce/rtpr/RtprHandler.kt

```diff
@@ -0,0 +1,175 @@
+
+    override fun start() {
+        config().getString("time-suffix")?.let {
+            timeSuffix = DateTimeFormatter.ofPattern(it)
+        }
+
+        vertx.eventBus().localConsumer<Packet>(RoutesCE.rtpr) { event ->
+            try {
+                val packet = event.body()
+                handle(packet)
+            } catch (e: Exception) {
+                logger.error("RtprHandler 'handle()' failed.", e)
+            }
+        }
+    }
+
+    open fun handle(packet: Packet) {
+        val payload = Unpooled.wrappedBuffer(packet.payload)
+        val report = RtpReportPayload().apply { decode(payload) }
+        val prefix = prefix(report.source)
+
+        writeToDatabase(prefix, packet, report)
+        calculateMetrics(prefix, packet, report)
+    }
+
+    open fun prefix(source: Byte): String {
+        return when (source) {
+            RtpReportPayload.SOURCE_RTP -> PREFIX_RTP
+            RtpReportPayload.SOURCE_RTCP -> PREFIX_RTCP
+            else -> throw IllegalArgumentException("Unsupported RTP report source: '${source}'")
+        }
+    }
+
+    open fun calculateMetrics(prefix: String, packet: Packet, report: RtpReportPayload) {
+        // Skip cumulative reports
+        if (report.cumulative) {
+            return;
+        }
+
+        val srcHost = packet.srcAddr.host
+        val dstHost = packet.dstAddr.host
+
+        report.apply {
+            val codecTag = codecName ?: payloadType.toString()
+            record(srcHost, dstHost, codecTag, prefix + EXPECTED_PACKETS, expectedPacketCount)
+            record(srcHost, dstHost, codecTag, prefix + LOST_PACKETS, lostPacketCount)
+            record(srcHost, dstHost, codecTag, prefix + REJECTED_PACKETS, rejectedPacketCount)
+
+            record(srcHost, dstHost, codecTag, prefix + JITTER, avgJitter)
+
+            record(srcHost, dstHost, codecTag, prefix + R_FACTOR, rFactor)
+            record(srcHost, dstHost, codecTag, prefix + MOS, mos)
+
+            Timer.builder(prefix + DURATION)
+                    .apply {
+                        srcHost?.let { tag("src_host", it) }
+                        dstHost?.let { tag("dst_host", it) }
+                        tag("codec", codecTag)
+                    }
+                    .register(Metrics.globalRegistry)
+                    .record(duration.toLong(), TimeUnit.MILLISECONDS)
+        }
+    }
+
+    open fun record(srcHost: String?, dstHost: String?, codecTag: String, name: String, value: Number) {
+        DistributionSummary.builder(name)
+                .apply {
+                    srcHost?.let { tag("src_host", it) }
+                    dstHost?.let { tag("dst_host", it) }
+                    tag("codec", codecTag)
+                }
+                .register(Metrics.globalRegistry)
+                .record(value.toDouble())
+    }
+
+    open fun writeToDatabase(prefix: String, packet: Packet, report: RtpReportPayload) {
+        val collection = if (report.cumulative) {
```

Comment:
I would prefer to define this `index` and `raw` part of prefix before calling this method. Can we move it above?